test_settings.py

```diff
@@ -46,24 +46,29 @@
     'JWT_SECRET_KEY': 'test-key',
 
     'JWT_PUBLIC_SIGNING_JWK_SET': (
-        '{"keys": [{"kid": "BTZ9HA6K", "e": "AQAB", "kty": "RSA", "n": "o5cn3ljSRi6FaDEKTn0PS-oL9EFyv1pI7dRgffQLD1qf5D6'
-        'sprmYfWWokSsrWig8u2y0HChSygR6Jn5KXBqQn6FpM0dDJLnWQDRXHLl3Ey1iPYgDSmOIsIGrV9ZyNCQwk03wAgWbfdBTig3QSDYD-sTNOs3pc'
-        '4UD_PqAvU2nz_1SS2ZiOwOn5F6gulE1L0iE3KEUEvOIagfHNVhz0oxa_VRZILkzV-zr6R_TW1m97h4H8jXl_VJyQGyhMGGypuDrQ9_vaY_RLEu'
-        'lLCyY0INglHWQ7pckxBtI5q55-Vio2wgewe2_qYcGsnBGaDNbySAsvYcWRrqDiFyzrJYivodqTQ"}]}'
+        '{"keys": [{"use": "sig", "e": "AQAB", "kty": "RSA", "alg": "RS512", "n": "kuI1T8prOHqX7Q9ARgyi7DsyLGMEGOmhPefU'
```

Comment:
[request] Can you please describe what the role of these changes is? Do we need to switch from RSA-256 to 512 in order to make this upgrade? Is the `use` field now mandatory, or is this an opportunistic improvement of some kind? etc.

Reply:
### 1

We're not switching from `RSA-256` to `512` as we're already using `RSA-512` 

https://github.com/edx/edx-internal/search?q=%22ALGORITHM%22

### 2
These fields are added by the tool which I used to generate the test key-value pair of the key.
Note: Even if we remove these additional fields, the tests still pass

Reply:
1. Ah OK! Somewhere I'd gotten the impression that the current ones were RS256. (Probably I was confusing key sizes and signature sizes.)
2. If new keys are needed, we'll want to create the actual replacement keys for our deployed environments in a secure manner. Can you document a way of generating a new keypair, so an engineer can do it on their laptop? We'll need to verify that the new key works with the existing library, too; I can do that by starting a key rotation in the staging environment. We can then use the same method to create the test key here, so that we know it matches in format.

It would also be great to use a multiline string with the JWT displayed in indented, key-sorted JSON so that it will be easier to read the code (and any diffs, in the future.)

Reply:
I've updated the keys which I've generated using this script https://github.com/mumarkhan999/rsa-key-generator.
I've also tested these keys on the master branch as well. https://github.com/openedx/edx-drf-extensions/pull/315.
The keys are in intended and sorted JSON now.

Reply:
@timmc-edx I've tested the new format of keys on the following sandbox.
- https://newkey.sandbox.edx.org/
I updated the `yml configuration` files from the sandbox server to add the new keys. Things are working fine.

Reply:
OK, great! I ended up reworking that script a bit as https://github.com/openedx/edx-platform/pull/32025/files although, as the comment on that PR points out, this duplicates an existing management command. So that PR might not actually merge. :-)

I'll use that script to generate keys and rotate them in stage, prod, and edge (https://github.com/edx/edx-arch-experiments/issues/261) and after that we'll be able to merge these changes.

Reply:
(This might take a bit -- I'm working with SRE to clarify the instructions for rotating these keys safely.)

Reply:
We've updated the existing keys in our deployed environments (all except sandboxes, which work differently) using a script I added in https://github.com/openedx/edx-platform/pull/32089 and I've added that to the Release Notes. Since we're going to be upgrading existing keys rather than replacing them, I've made a PR against your branch that does the same to the existing test keys: https://github.com/openedx/edx-drf-extensions/pull/337